src/main.cpp

```diff
@@ -198,6 +202,30 @@ namespace {
     set<int> setDirtyFileInfo;
 } // anon namespace
 
+bool HaveThinblockNodeConnections()
+{
+    BOOST_FOREACH(string& strAddNode, mapMultiArgs["-connect-thinblock"]) {
+        if(FindNode(strAddNode)) {
+            return true;
+        }
+    }
+    return false;
+
+}
+
+bool IsThinBlocksEnabled() 
+{
+    return GetBoolArg("-use-thinblocks", true);
+}
+
+bool IsChainNearlySyncd() 
+{
+    LOCK(cs_main);
+    if(chainActive.Height() < pindexBestHeader->nHeight - 2)
+        return false;
+    return true;
+}
+
```

Comment:
To help subsequent rebases, I'd like to minimize changes to Core files.  In particular, please move every function definition to a non-core file.  If you don't have enough new code to make a new file, or is "random stuff" put your functions in unlimited.cpp, thanks!